jdaviz/app.py

```diff
@@ -887,6 +848,179 @@ def _get_all_subregions(mask, spec_axis_data):
 
         return regions
 
+    def get_subsets(self, subset_name=None, spectral_only=False,
+                    spatial_only=False, astropy_region=False):
```

Comment:
passing `astropy_region=True` within specviz raises a cryptic error:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [5], line 1
----> 1 specviz.app.get_subsets('Subset 1', astropy_region=True)

File ~/repos/jdaviz/jdaviz/app.py:911, in Application.get_subsets(self, subset_name, spectral_only, spatial_only, astropy_region)
    909 elif not spectral_only and not spatial_only:
    910     if astropy_region:
--> 911         all_subsets[label] = subset_region[0]['region']
    912     else:
    913         all_subsets[label] = subset_region

File ~/repos/specutils/specutils/spectra/spectral_region.py:174, in SpectralRegion.__getitem__(self, item)
    172     return self.__getslice__(item)
    173 else:
--> 174     return SpectralRegion([self._subregions[item]])

TypeError: list indices must be integers or slices, not str
```

Reply:
Should be an easy fix, thanks for pointing it out!

Reply:
If not already, a test should be added to cover this use case.